src/main/java/net/javadiscord/javabot/systems/help/forum/ForumHelpManager.java

```diff
@@ -0,0 +1,140 @@
+	 */
+	public ReplyCallbackAction replyHelpThanks(IReplyCallback callback, @NotNull List<Member> helpers) {
+		List<ItemComponent> helperThanksButtons = new ArrayList<>(20);
+		for (Member helper : helpers.subList(0, Math.min(helpers.size(), 20))) {
+			helperThanksButtons.add(Button.success(ComponentIdBuilder.build(HELP_THANKS_IDENTIFIER, postThread.getId(), helper.getId()), helper.getEffectiveName())
+					.withEmoji(Emoji.fromUnicode("❤"))
+			);
+		}
+		ActionRow controlsRow = ActionRow.of(
+				Button.primary(ComponentIdBuilder.build(HELP_THANKS_IDENTIFIER, postThread.getId(), "done"), "I'm done here. Close this post!"),
+				Button.danger(ComponentIdBuilder.build(HELP_THANKS_IDENTIFIER, postThread.getId(), "cancel"), "Cancel Closing")
+		);
+		List<ActionRow> rows = new ArrayList<>();
+		rows.add(controlsRow);
+		rows.addAll(MessageActionUtils.toActionRows(helperThanksButtons));
+		return callback.reply(THANK_MESSAGE_TEXT)
+				.setComponents(rows);
+	}
+
+	/**
+	 * Closes the {@link ForumHelpManager#postThread}.
+	 *
+	 * @param callback    The callback to reply to.
+	 * @param withHelpers Whether the help-thanks message should be displayed.
+	 * @param reason      The reason for closing this post.
+	 */
+	public void close(IReplyCallback callback, boolean withHelpers, @Nullable String reason) {
+		List<Member> helpers = getPostHelpers();
+		if (withHelpers && !helpers.isEmpty()) {
+			replyHelpThanks(callback, helpers).queue();
+			return;
+		}
+		Responses.info(callback, "Post Closed", "This post has been closed by %s%s", callback.getUser().getAsMention(), reason != null ? " for the following reason:\n> " + reason : ".")
+				.setEphemeral(false)
+				.queue(s -> postThread.getManager().setLocked(true).setArchived(true).queue());
+	}
+
+	/**
+	 * Thanks a single user.
+	 *
+	 * @param event      The {@link ButtonInteractionEvent} that was fired.
+	 * @param postThread The {@link ThreadChannel} post.
+	 * @param helperId   The helpers' discord id.
+	 */
+	public void thankHelper(@NotNull ButtonInteractionEvent event, ThreadChannel postThread, long helperId) {
+		event.getJDA().retrieveUserById(helperId).queue(helper -> {
+			// First insert the new thanks data.
+			try {
+				DbActions.update(
+						"INSERT INTO help_channel_thanks (reservation_id, user_id, channel_id, helper_id) VALUES (?, ?, ?, ?)",
+						postThread.getIdLong(),
+						postThread.getOwnerIdLong(),
+						postThread.getIdLong(),
+						helper.getIdLong()
+				);
+				HelpConfig config = Bot.getConfig().get(event.getGuild()).getHelpConfig();
+				HelpExperienceService service = new HelpExperienceService(Bot.getDataSource());
+				// Perform experience transactions
+				service.performTransaction(helper.getIdLong(), config.getThankedExperience(), HelpTransactionMessage.GOT_THANKED, event.getGuild());
+				service.performTransaction(postThread.getOwnerIdLong(), config.getThankExperience(), HelpTransactionMessage.THANKED_USER, event.getGuild());
```

Comment:
Should the OP really get experience for every helper they thanked?

Reply:
Well, it was always like that. Its just 3 XP, so, well. I think we can even get rid of that